development/Dockerfile

```diff
@@ -1,40 +1,78 @@
+FROM ghcr.io/nautobot/nautobot-dev:${NAUTOBOT_VER}-py${PYTHON_VER}
+
+# Runtime argument and environment setup
 ARG NAUTOBOT_ROOT=/opt/nautobot
 
+ENV prometheus_multiproc_dir=/prom_cache
 ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
 
-WORKDIR $NAUTOBOT_ROOT
-
-# Configure poetry
-RUN poetry config virtualenvs.create false \
-    && poetry config installer.parallel false
+# Install Poetry manually via its installer script;
+# We might be using an older version of Nautobot that includes an older version of Poetry
+# and CI and local development may have a newer version of Poetry
+# Since this is only used for development and we don't ship this container, pinning Poetry back is not expressly necessary
+# We also don't need virtual environments in container
+RUN curl -sSL https://install.python-poetry.org | python3 - && \
+  poetry config virtualenvs.create false
 
+# !!! USE CAUTION WHEN MODIFYING LINES ABOVE
 # -------------------------------------------------------------------------------------
-# Install Nautobot Plugin
+# App-specifc system build/test dependencies.
+# 
+# Example: LDAP requires `libldap2-dev` to be apt-installed before the Python package.
 # -------------------------------------------------------------------------------------
-WORKDIR /tmp
+# --> Start safe to modify section
 
-# Copy in only pyproject.toml/poetry.lock to help with caching this layer if no updates to dependencies
-COPY poetry.lock pyproject.toml /tmp/
-# --no-root declares not to install the project package since we're wanting to take advantage of caching dependency installation
-# and the project is copied in and installed after this step
-RUN poetry self update && \
-    poetry add nautobot=$NAUTOBOT_VER && \
-    poetry update --lock && \
-    poetry install --no-interaction --no-ansi --no-root
+# Uncomment the lines below if you are apt-installing any package.
+# RUN apt-get -y update && apt-get -y install \
+#     libldap2-dev \
+#     && rm -rf /var/lib/apt/lists/*
 
-WORKDIR /source
+# --> Stop safe to modify section
+# -------------------------------------------------------------------------------------
+# Install Nautobot App
+# -------------------------------------------------------------------------------------
+# !!! USE CAUTION WHEN MODIFYING LINES BELOW
 
-# Copy in the rest of the source code and install local Nautobot plugin
+# Copy in the source code
+WORKDIR /source
 COPY . /source
-RUN mv /tmp/poetry.lock /tmp/pyproject.toml /source && \
-    poetry install --no-interaction --no-ansi
+
+# Get container's installed Nautobot version as a forced constraint
+# NAUTOBOT_VER may be a branch name and not a published release therefor we need to get the installed version
+#  so pip can use it to recognize local constraints.
+RUN pip show nautobot | grep "^Version: " | sed -e 's/Version: /nautobot==/' > constraints.txt
+
+# Use Poetry to grab dev dependencies from the lock file
+# Can be improved in Poetry 1.2 which allows `poetry install --only dev`
```

Comment:
Mostly a question to @bryanculver but since we're now explicitly installing latest Poetry, can we make this improvement now?

Reply:
Sure? But let's not do this one-off just here, let's keep it consistent for all other apps at the moment and use drift management. This is the intended purpose of cookie versions.

Reply:
https://github.com/networktocode-llc/cookiecutter-ntc/issues/327